Tests/StreamChatTests/Database/DTOs/ChannelDTO_Tests.swift

```diff
@@ -614,6 +614,133 @@ final class ChannelDTO_Tests: XCTestCase {
                 .map(\.user.id)
         )
     }
+    
+    func test_lastMessageFromCurrentUser() throws {
+        let user: UserPayload = dummyCurrentUser
+        let channelId: ChannelId = .unique
+        let message1 = MessagePayload(
+            id: .unique,
+            type: .regular,
+            user: user,
+            createdAt: Date.distantPast,
+            updatedAt: .unique,
+            deletedAt: nil,
+            text: .unique,
+            showReplyInChannel: true,
+            mentionedUsers: [dummyCurrentUser],
+            replyCount: 1,
+            extraData: [:],
+            reactionScores: ["like": 1],
+            reactionCounts: ["like": 1],
+            isSilent: false,
+            isShadowed: false,
+            attachments: [],
+            pinned: false
+        )
+        
+        let message2 = MessagePayload(
+            id: .unique,
+            type: .regular,
+            user: user,
+            createdAt: Date(),
+            updatedAt: .unique,
+            deletedAt: nil,
+            text: .unique,
+            showReplyInChannel: false,
+            mentionedUsers: [dummyCurrentUser],
+            replyCount: 0,
+            extraData: [:],
+            reactionScores: ["like": 1],
+            reactionCounts: ["like": 1],
+            isSilent: false,
+            isShadowed: false,
+            attachments: [],
+            pinned: false
+        )
+        
+        let channel = dummyPayload(with: channelId, messages: [message1, message2])
```

Comment:
Would be nice to add ephemeral messages, and messages which are not sent, to verify the 2 new predicates we added 👍

Reply:
Added